### ISSUE TITLE
sandbox, navigator: Options to pick a free port automatically, and report resolved port number.

### ISSUE THREAD

**Issue by associahedron** [digital-asset/daml#833]
As part of making `daml start` work when called multiple times (see issue #764), and without introducing race conditions (see PR #824), we would like Sandbox and Navigator to have options that would let them pick a free port themselves (e.g. by using port 0, which resolves to free port automatically), and to report that port to the parent process in some way.

Current state:
* Sandbox already accepts `--port 0`, but the resolved port number is buried among all the other output.
* Navigator also accepts `--port 0`, but the resolved port number doesn't appear in the output.

My suggestion: both of these tools should take an extra option `--report-port PATH` that causes them to write to the specified file the resolved port number, once the server is ready. (This would also solve another issue with `daml start`, which is knowing when the servers are ready to serve.) I'm open to other suggestions though.

cc @rautenrieth-da @moritzkiefer-da @akalburgi

**Comment by gleber-da** [comment]
@associahedron @hurryabit FYI this almost identical mechanism which we use in couple of integration tests:
https://github.com/digital-asset/daml/blob/d495428ccf8a9132c3e9086eff4b793bb75c8824/bazel_tools/client_server_test/client_server_test.bzl#L65

**Comment by mziolekda** [comment]
Implemented
